src/Microsoft.Azure.WebJobs.Host/Executors/JobHostContextFactory.cs

```diff
@@ -188,8 +188,7 @@ public JobHostContextFactory(IStorageAccountProvider storageAccountProvider, ICo
                         queueConfiguration, exceptionHandler, trace, functions,
                         functionInstanceLogger, functionExecutor);
 
-                    Guid hostInstanceId = Guid.NewGuid();
```

Comment:
Note that technically this will be a breaking change and would affect anyone trying to run multiple instances of a JobHost on a single machine. Not sure how prevalent that is, but in such cases you'd have two separate instances using the same lease ID, and potentially other collisions where we assumed instance IDs were globally unique before.

I do think that spinning up multiple hosts on the same machine is an anti pattern, and the benefits of this change likely outweigh these concerns. I just want to make sure we're not missing anything here.

@MikeStall ?